### ISSUE TITLE
Introduce gradient descent, learning rate, and batch-size together

### ISSUE THREAD

**Issue by dafnevk** [carpentries-incubator/deep-learning-intro#106]
Episode 1 and 2 shortly mention the optimizer, but do not get into detail on how they work. Episode 3 quickly introduces batch size, but not yet learning rate.
These concepts are all related and would probably benefit from being explained together. They become relevant in episode 3 when we try to improve the training process.

Maybe when we split episode 3 into two parts, the second part could start with some theory on optimization, together with some diagrams.

Callouts:
- How to avoid getting stuck in a local minimum?

**Comment by psteinb** [comment]
I like this idea. I felt, that we had quite a AHA moment, when we shared how the optimisation approach chosen with DL today, enables us to consume much more data that can possibly be done with tradition global optimisation.

**Comment by florian-huber** [comment]
Fully agree as well. It would be really good to introduce those terms more properly (and simply adding them all to episode 3 would make it too much I believe).

**Comment by svenvanderburg** [comment]
Relates to #251
